src/System.Runtime/tests/System/Reflection/AssemblyTests.cs

```diff
@@ -197,8 +197,16 @@ public static void GetModules_GetModule(Assembly assembly)
             Assert.NotEmpty(assembly.GetModules());
             foreach (Module module in assembly.GetModules())
             {
-                Assert.NotNull(module);
-                Assert.Equal(module, assembly.GetModule(module.Name));
```

Comment:
In this case we are performing test for type containing List<T> class. This is in System.Collections.dll for .net core and in mscorlib.dll for desktop. For single module assembly typically module name is same is file name. But for mscorlib.dll it is specifically specified as CommonLanguageRuntimeLibrary in IL. module.Name property returns the name of the module with the path removed. However in this test we want to find the name of module and not path name to provide as input to assembly.GetModule(). So using module.Name to get that is wrong. module.ToString is the correct api which provides the name of module. So you should just replace all instances of
module.Name
with 
module.ToString()
in this file